python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_cleaning_strategy.py

```diff
@@ -0,0 +1,588 @@
+    g_function : ot.Function
+        The function.
+    input_distribution : ot.Distribution
+        The input distribution.
+    n_valid : int
+        The number of simulations to compute the Q2 score.
+    
+    Returns
+    -------
+    Q2 : float
+        The Q2 score
+    """
+    metamodel = polynomialchaos_result.getMetaModel()
+    inputTest = input_distribution.getSample(n_valid)
+    outputTest = g_function(inputTest)
+    val = ot.MetaModelValidation(inputTest, outputTest, metamodel)
+    Q2 = val.computePredictivityFactor()[0]
+    graph = val.drawValidation()
+    graph.setTitle("Q2=%.2f%%" % (Q2 * 100))
+    view = otv.View(graph, figure_kw={"figsize": (5.0, 4.0)})
+    return view
+
+
+# %%
+#
+# We consider the Ishigami model which has three inputs uniform in the :math:`[-\pi, \pi]` interval. This is an interesting example for our purpose because it is highly non linear, so that a high polynomial degree will be required in order to produce a polynomial chaos expansion with Q2 score sufficiently close to 1.
+
+# %%
+im = ishigami_function.IshigamiModel()
+
+im.distributionX.setDescription(["`X_0`", "`X_1`", "`X_2`"])
+im.model.setOutputDescription(["`Y`"])
+print(im.distributionX)
+
+# %%
+#
+# Then we create the multivariate basis onto which the function is expanded. By default, it is associated with the linear enumeration rule. Since our marginals are uniform, the `OrthogonalProductPolynomialFactory` class produce Legendre polynomials.
+# In order to create the multivariate basis of polynomials, we must specify the number of functions in the basis. In this particular case, we compute that number depending on the total degree. The `getMaximumDegreeStrataIndex` method of the enumeration function computes the number of layers necessary to achieve that total degree. Then the number of functions up to that layer is computed with the `getStrataCumulatedCardinal` method.
+
+# %%
+dimension = im.distributionX.getDimension()
+multivariateBasis = ot.OrthogonalProductPolynomialFactory(
+    [im.distributionX.getMarginal(i) for i in range(dimension)]
+)
+
+totalDegree = 5  # Polynomial degree
+enumerate_function = multivariateBasis.getEnumerateFunction()
+strataIndex = enumerate_function.getMaximumDegreeStrataIndex(totalDegree)
+print("strataIndex = ", strataIndex)
+number_of_terms_in_basis = enumerate_function.getStrataCumulatedCardinal(strataIndex)
+print("number_of_terms_in_basis = ", number_of_terms_in_basis)
+adaptiveStrategy = ot.FixedStrategy(multivariateBasis, number_of_terms_in_basis)
+print(adaptiveStrategy)
+
+# %%
+#
+# We compute the coefficients using a multivariate tensor product Gaussian quadrature rule. Since the coefficients are computed in the standardized space, we first use the `getMeasure` method of the multivariate basis in order to get that standardized distribution. Then we use the `GaussProductExperiment` class to create the quadrature, using 6 nodes on each of the dimensions.
```

Comment:
```suggestion
# We compute the coefficients using a multivariate tensor product Gaussian quadrature rule.
# Since the coefficients are computed in the standardized space, we first use the :meth:`~openturns.OrthogonalProductPolynomialFactory.getMeasure` method of the multivariate basis in order to get that standardized distribution.
# Then we use the :class:`~openturns.GaussProductExperiment` class to create the quadrature, using 6 nodes on each of the dimensions.
```